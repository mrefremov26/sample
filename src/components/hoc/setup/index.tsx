import React from 'react';

import { withAuthChecker } from '@hoc/withAuthChecker';
import { withModals } from '@hoc/withModals';
import { withProviders } from '@hoc/withProviders';
import { withScrollReset } from '@hoc/withScrollReset';

export const setup = (Component: React.FC<any>): any => {
  return compose(Component, React.memo, withScrollReset, withAuthChecker, withModals, withProviders);
};

function compose(Component: React.FC, ...functions: Array<(...args: any[]) => React.FC>) {
  return functions.reduce((acc: React.FC, f: (...args: any[]) => React.FC) => f(acc), Component);
}
