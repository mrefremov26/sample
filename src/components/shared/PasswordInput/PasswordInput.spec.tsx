import * as React from 'react';
import renderer from 'react-test-renderer';

import { PasswordInput } from '.';
import { setup } from '@hoc/setup';

describe('PasswordInput renders correctly', () => {
  it('snapshot', () => {
    const Component = setup(PasswordInput);
    const tree = renderer.create(<Component />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
