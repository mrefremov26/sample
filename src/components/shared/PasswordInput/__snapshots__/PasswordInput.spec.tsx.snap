// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput renders correctly snapshot 1`] = `
Array [
  <div>
    <input
      className=""
      name="input"
      type="password"
    />
    <div>
      <div
        onClick={[Function]}
      >
        <div />
      </div>
    </div>
  </div>,
  " ",
]
`;
